### PR TITLE
resource score

### PR DESCRIPTION
**Pull request by Gordon5K**
Not random, but a function to get a score between min and max resource based on the position of a point within the radius.

min and maxResource score functions will need to be swapped out for global constants.

### CODE REVIEW

File: server/services/random.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
+const minResourceScore = 10;
+const maxResourceScore = 50;
```
Comment (T4G2) [inline server/services/random.js:2]: constants should be written in CAPITAL_LETTERS_WITH_UNDERCASE

File: server/services/random.js [hunk 2]
```diff
@@ -33,4 +36,17 @@ module.exports = class RandomService {
...
     }
     
+
+    getResourceScore(radius, x, y){
+        let vector = Math.hypot(x, y);
+        
+        //How far from the outside (%) is the point
+        let vectorScale = (radius - vector)/radius;
+
+        let resourceRange = maxResourceScore - minResourceScore;
```
Comment (T4G2) [inline server/services/random.js:46]: resource range can be constant (look up)

File: server/services/random.js [hunk 3]
```diff
@@ -1,3 +1,6 @@
+const minResourceScore = 10;
+const maxResourceScore = 50;
+
```
Comment (T4G2) [inline server/services/random.js:3]: also we can add const resourceRange in here nad i would rename it to RESOURCE_DLETA (delta is almost everytime in math max - min)

File: server/services/random.js [hunk 4]
```diff
@@ -33,4 +36,17 @@ module.exports = class RandomService {
...
+    getResourceScore(radius, x, y){
+        let vector = Math.hypot(x, y);
+        
+        //How far from the outside (%) is the point
+        let vectorScale = (radius - vector)/radius;
+
+        let resourceRange = maxResourceScore - minResourceScore;
+
+        let score = minResourceScore + (resourceRange * vectorScale);
```
Comment (T4G2) [inline server/services/random.js:48]: i suggest this to be : 

```
let score = MIN_RESOURCE + (RESOURCE_DELTA * Math.pow(RS_BASE, - RS_EXPONENT * Math.pow(vectorScale, 2)));
```

File: server/services/random.js (hunk 3 above)
Comment (T4G2) [reply server/services/random.js:3]: Also add here constants RS_BASE and RS_EXPONENT 

```
const RS_BASE = 2
const RS_EXPONENT = 5.8
```

File: server/services/random.js (hunk 1 above)
Comment (mike-eason) [reply server/services/random.js:2]: These look like placeholder values. Remove them and instead use parameters on the function.

File: server/services/random.js [hunk 5]
```diff
@@ -33,4 +36,17 @@ module.exports = class RandomService {
         return position;
     }
     
+
+    getResourceScore(radius, x, y){
```
Comment (mike-eason) [inline server/services/random.js:40]: Add parameters for min and max resources.

Comment (T4G2) [reply server/services/random.js:40]: @mike-eason  i think it is alright if is as constant

File: server/spec/random.spec.js [hunk 6]
```diff
@@ -47,4 +47,68 @@ describe('random', () => {
         expect(res.y).toBeLessThanOrEqual(radius);
     });
 
+
+    describe('getResourceScore', () => {
+
+        const minResourceValue = 10;
+        const maxResourceValue = 50;
```
Comment (T4G2) [inline server/spec/random.spec.js:54]: duplicated const

**Pull request merged**